device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ReportedPropertiesParameters.java

```diff
@@ -0,0 +1,35 @@
+package com.microsoft.azure.sdk.iot.device;
+
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.Property;
+
+import java.util.Set;
+
+/**
+ * Convenience class for the sendReportedProperties method in the DeviceClient
+ */
+public class ReportedPropertiesParameters {
```

Comment:
Could this be a builder class?

Reply:
And typically in Java, we don't make public fields, but do expose public getters/setters. Lombok can help you with that. [Here](https://github.com/Azure/azure-iot-sdk-java/blob/master/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClientOptions.java) is a good example of a similar class that uses Lombok to generate the getters/setters/builder class with minimal code

Reply:
Check the update class. I forced the usage to have parameters and then you can set the callbacks individually.

Reply:
Interesting, I wasn't aware of Lombok.